integration-tests/src/integrationTest/org/apache/cassandra/sidecar/routes/CompactionStopIntegrationTest.java

```
@@ -0,0 +1,218 @@
+    private static final QualifiedName TEST_TABLE = new QualifiedName(TEST_KEYSPACE, TEST_TABLE_PREFIX + "_compaction_test");
+
+    @Override
+    protected void initializeSchemaForTest()
+    {
+        createTestKeyspace(TEST_KEYSPACE, DC1_RF1);
+        createTestTable(TEST_TABLE,
+                        "CREATE TABLE %s ( \n" +
+                        "  id int PRIMARY KEY, \n" +
+                        "  data text \n" +
+                        ");");
+    }
+
+    @Override
+    protected void beforeTestStart()
+    {
+        // Wait for schema initialization
+        waitForSchemaReady(30, TimeUnit.SECONDS);
+    }
+
+    @Test
+    void testStopCompactionByTypeSuccess()
+    {
+        // Create SSTables
+        insertTestData(TEST_TABLE, 1000);
+        cluster.stream().forEach(instance -> instance.flush(TEST_KEYSPACE));
+
+        // Trigger non-blocking compaction in background
+        cluster.stream().forEach(instance -> instance.nodetool("compact", TEST_KEYSPACE));
+
+        // Call compaction stop endpoint
+        String payload = "{\"compaction_type\":\"COMPACTION\"}";
+        HttpResponse<Buffer> response = getBlocking(
+            trustedClient()
+                .post(serverWrapper.serverPort, "localhost", COMPACTION_STOP_ROUTE)
+                .sendBuffer(buffer(payload))
+                .expecting(HttpResponseExpectation.SC_OK)
+        );
+
+        assertThat(response.statusCode()).isEqualTo(HttpResponseStatus.OK.code());
+        CompactionStopResponse stopResponse = response.bodyAsJson(CompactionStopResponse.class);
+        assertThat(stopResponse).isNotNull();
+        assertThat(stopResponse.status()).isEqualTo("SUCCESS");
+        assertThat(stopResponse.compactionType()).isEqualTo("COMPACTION");
+        assertThat(stopResponse.errorCode()).isEqualTo("200 OK");
+        assertThat(stopResponse.reason()).isEqualTo("Operation Succeeded");
+    }
+
+    @Test
+    void testStopCompactionByIdSuccess()
```

In the current implementation, a request with a non-existent compaction ID silently succeeds. The JMX call simply succeeds without any action being taken. It would be more informative to update the Cassandra JMX implementation to return a boolean value or some other indicator. I will seek input from others on this.

https://github.pie.apple.com/slaha/aci-cassandra/blob/38ab7d02e0979074f3de1c7f90ce70668d2492b9/src/java/org/apache/cassandra/db/compaction/CompactionManager.java#L2306